script.js

```diff
@@ -1,4 +1,251 @@
+//--- GLOBAL VARIABLES ---
+
+var gameMode = "startGame"; // startGame, playerTurn, computerTurn
+var cardDeck = [];
+var dealerHand = [];
+var dealerScore = 0;
+var playerHand = [];
+var playerScore = 0;
+
+//--- HELPER FUNCTIONS ---
+
+function makeDeck() {
+  var cardDeck = [];
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+      cardDeck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+}
+
+function getRandomIndex(max) {
+  return Math.floor(Math.random() * max);
+}
+
+function shuffleCards(cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+}
+
+function dealCards(handArray, cardDeck, noOfCards) {
+  for (var counter = 0; counter < noOfCards; counter += 1) {
+    handArray.push(cardDeck.pop());
+  }
+  return handArray;
+}
+
+function isBlackjack(handArray) {
+  return (
+    (handArray[0].name === "ace" && handArray[1].rank >= 10) ||
+    (handArray[1].name === "ace" && handArray[0].rank >= 10)
+  );
+}
+
+function calculateScore(handArray) {
+  var score = 0;
+  for (var counter = 0; counter < handArray.length; counter += 1) {
+    score = score + getCardScore(handArray[counter]);
+  }
+  //if there are aces, update ace value in score. if total hand value is >21, update ace value from 11 to 1
+  for (var counter = 0; counter < handArray.length; counter += 1) {
+    if (handArray[counter].name === "ace") {
+      score = updateAceValueinScore(score);
+    }
+  }
+  return score;
+}
+
+function getCardScore(card) {
```

Comment:
Rather than having a function to get the score, you could modify the makeDeck function to have the score as a key.